oolab/src/main/java/agh/ics/oop/model/AbstractWorldMap.java

```diff
@@ -0,0 +1,79 @@
+package agh.ics.oop.model;
+
+import agh.ics.oop.model.util.MapVisualizer;
+
+import java.util.*;
+
+public abstract class AbstractWorldMap implements WorldMap {
+    protected final Set<Vector2d> occupiedPositions = new HashSet<>();
```

Comment:
trochę mam problem z tą koncepcją - wydaje się, że to zalążek nietypowego, ale niekoniecznie złego pomysłu, ale nie został w ogóle pociągnięty do końca. Używamy tego tylko dla zwierząt w jakiś sposób rozdzielając kolekcję pozycji od samych zwierzaków (czyli to co wcześniej było hashmapą i w sumie dalej mogłoby być?), a w zamian nic nie dostajemy. Nie używa Pan tego przy trawach, więc one nigdy nie są w zajmowanych pozycjach.